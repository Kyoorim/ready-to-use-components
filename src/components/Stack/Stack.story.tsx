import React from 'react';
import { Stack } from './Stack';

export default {
  title: 'Stack',
  component: Stack,
};

export const Default = () => {
  return <Stack></Stack>;
};
