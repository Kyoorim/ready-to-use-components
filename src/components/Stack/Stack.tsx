import React from 'react';
import styled from 'styled-components';
// eslint-disable-next-line prettier/prettier
import { flexbox, FlexboxProps, grid, GridProps, layout, LayoutProps } from 'styled-system';

const StackBox = styled.div`
  ${flexbox}
  ${grid}
  ${layout}
`;

export type StackProps = FlexboxProps & GridProps & LayoutProps;

const Stack: FCC<StackProps> = (props: any) => {
  return <StackBox {...props} data-testid="stack-comp"></StackBox>;
};

export default Stack;
