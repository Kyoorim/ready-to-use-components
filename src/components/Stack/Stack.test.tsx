import React from 'react';
import { render, screen } from '@testing-library/react';
import Stack from './Stack';

const defaultProps = {
  onClick: jest.fn(),
  children: 'Stack',
};

describe('<Stack />', () => {
  it('should render', () => {
    render(<Stack {...defaultProps} />);

    expect(screen.getByTestId('stack-comp')).toBeInTheDocument();
  });

  it('should render children', () => {
    render(<Stack {...defaultProps} />);

    expect(screen.getByText(/Stack/i)).toBeInTheDocument();
  });

  it('should allow custom className', () => {
    const props = {
      ...defaultProps,
      className: 'Custom',
    };
    render(<Stack {...props} />);

    expect(screen.getByTestId('stack-comp')).toHaveProperty('className');
  });
});
