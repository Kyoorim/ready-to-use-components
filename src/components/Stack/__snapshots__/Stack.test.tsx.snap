// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stack /> should render 1`] = `
<div
  class="sc-bcXHqe csJgvw"
  data-testid="stack-comp"
>
  Stack
</div>
`;
